src/SprykerSdk/Zed/AppSdk/Business/AsyncApi/Builder/AsyncApiBuilder.php

```diff
@@ -74,6 +77,13 @@ public function addAsyncApiMessage(AsyncApiRequestTransfer $asyncApiRequestTrans
         $asyncApi = Yaml::parseFile($targetFile);
 
         $asyncApiMessageTransfer = $asyncApiRequestTransfer->getAsyncApiMesssageOrFail();
+
+        if (($asyncApiMessageTransfer->getPayloadTransferObjectName() === null && count($asyncApiMessageTransfer->getProperty()) === 0) || ($asyncApiMessageTransfer->getPayloadTransferObjectName() !== null && count($asyncApiMessageTransfer->getProperty()) > 0)) {
+            throw new NullValueException(
```

Comment:
The exception name is not ideal. How about `InvalidConfigurationException`? You can also propose other names if you like.